src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -30,63 +30,173 @@
 
 namespace Microsoft.CodeAnalysis.UnitTests
 {
-    [CollectionDefinition(Name)]
-    public class AssemblyLoadTestFixtureCollection : ICollectionFixture<AssemblyLoadTestFixture>
-    {
-        public const string Name = nameof(AssemblyLoadTestFixtureCollection);
-        private AssemblyLoadTestFixtureCollection() { }
-    }
+#if NETCOREAPP
 
-    public sealed class DefaultAnalyzerAssemblyLoaderTests : TestBase
+    public sealed class InvokeUtil
     {
-#if NETCOREAPP
-        private sealed class InvokeUtil
+        public void Exec(Action<string> testOutputHelper, AssemblyLoadContext alc, bool shadowLoad, string typeName, string methodName)
         {
-            public void Exec(string typeName, string methodName) => InvokeTestCode(typeName, methodName);
+            // Ensure that the test did not load any of the test fixture assemblies into 
+            // the default load context. That should never happen. Assemblies should either 
+            // load into the compiler or directory load context.
+            //
+            // Not only is this bad behavior it also pollutes future test results.
+            var count = AssemblyLoadContext.Default.Assemblies.Count();
+            using var fixture = new AssemblyLoadTestFixture();
+            using var tempRoot = new TempRoot();
+            var loader = shadowLoad
+                ? new ShadowCopyAnalyzerAssemblyLoader(alc, tempRoot.CreateDirectory().Path)
```

Comment:
I'm missing a bit of context. When is the Core+ShadowCopy combination useful? I though shadow copying was mostly a workaround for the locking that Desktop does when loading assemblies.

Reply:
The primary purpose of shadow copy is to avoid locking files on disk. Analyzers are frequently loaded from locations that developers delete as a part of normal dev operations. Hence we shadow copy to allow normal dev operations to occur. 

That is true for both .NET Framework and .NET Core. Experimentally .NET Core + Linux allows delete without shadow copy but we're unsure if that is 100% viable for our scenarios. Part of #66532 is discussing that with the runtime team.